examples/PKS0637-752.jl

```diff
@@ -3,27 +3,29 @@
 using SyncSSCModel
 using SpectralFitting
 using Plots
+using Revise
 
 # define the model we want to fit
 # includes default parameter values
 # specifies which paramters are free
 function PKSSSCModel(;
-    K = FitParam(1.0),
-    B = FitParam(1.25E-6),
-    n_e0 = FitParam(19.0),
-    radius = FitParam(1.0E22),
+    K = FitParam(1.0, lower_limit = 0.5, upper_limit = 2.0),
+    B = FitParam(1.25E-5, lower_limit = 0.0, upper_limit = 1.0E-3),
+    n_e0 = FitParam(19.0, lower_limit = 0.0, upper_limit = 1.0E2),
+    radius = FitParam(1.0E22, lower_limit = 1.0E21, upper_limit = 1.0E23),
     Γ = FitParam(2.0),
     γ_min = FitParam(2.5E3),
     γ_max = FitParam(4.0E6),
-    p = FitParam(2.6),
-    dL = FitParam(1.26E28),
+    p = FitParam(2.6, lower_limit = 2.0, upper_limit = 3.5),
+    dL = FitParam(1.26E28, lower_limit = 1.0E27, upper_limit = 1.0E29),
     θ = FitParam(60.0 * pi / 180.0),
     z = FitParam(0.651),
 )
     SSCModel{
         typeof(K),
         # SpectralFitting.FreeParameters{(:K, :B, :p, :radius, :θ, :dL, :z, :n_e0)},
-        SpectralFitting.FreeParameters{(:K, :B, :p, :θ, :n_e0,)},
+        # SpectralFitting.FreeParameters{(:K, :B, :p, :θ, :n_e0,)},
+        SpectralFitting.FreeParameters{(:B, :n_e0)},
```

Comment:
Is there a reason why `K` isn't a free parameter for PKS0637-752, but it is for Pictor A?

Reply:
Ideally `K` would be fixed at 1 for all of the fits. The "normalisation" should change as, e.g., the luminosity distance `dL` is changed or `radius`. However, the fitting algorithm can't find the best fit value for `dL`, for example - maybe because the step size is too large (?) Not sure why that would be the case though because the algorithm isn't using an absolute step size. If `K` is a free parameter it can be fit for. This is OK for Pictor A because the fit parameters start off close to the best values and `K` changes to still be $\mathcal{O}(1)$. But for PKS 0637-752 $K \rightarrow \infty$ (or at least the upper bound) if it is free which isn't sensible. An additional complication which affects the physics not the fitting is that the model can't properly explain the PKS spectrum because the high frequency cut-off of the synchrotron component is not realistic.

Reply:
Ah I see. I'll have a go at fitting `dL` over the weekend and see if there's an obvious issue with the algorithm. My first instinct is to have a look at whether the artificial covariance we're prescribing is in someway distorting the fit? Have you tried increasing/decreasing the variance of the data?

Reply:
The problem seems to be the wildly different dynamical ranges, e.g., $K \in [0, 2]$ while $d_L \in [10^{27}, 10^{39}]$. Fitting for $\log_{10} d_L$ instead works fine. Perhaps rescaling all of the parameters so we're essentially fitting over similar ranges might be required. This might indeed be a problem with the covariances. Might have to look more "under the hood" at the implementation of `LevenbergMarquadt`.

But for now let's keep this Pull request as a draft (can I update that?) and I'll change some parameters to be in log space.

Reply:
Interesting that $\log d_L$ fits well. The algorithm should be fairly robust against magnitudes, and the covariance is for the data not directly for the parameters. Will need to investigate a little.

Reply:
As far as I remember, the luminosity distance dL should be fixed for each object and shouldn't vary unless the redshift changes. dL is only as a function of the speed of light c, the redshift z and the Hubble parameter Ho, according to Dermer et al. 1997 paper.